reactive-banana/src/Reactive/Banana/Combinators.hs

```diff
@@ -277,12 +277,13 @@ observeE = E . Prim.observeE . Prim.mapE unM . unE
 -- | Dynamically switch between 'Event'.
 -- Semantically,
 --
--- > switchE ee = \time0 -> concat [trim t1 t2 e | (t1,t2,e) <- intervals ee, time0 <= t1]
+-- > switchE e0 ee = \time0 -> [(t, a) | (t,a) <- e0, t < t1] ++ concat [trim t1 t2 e | (t1,t2,e) <- intervals ee, time0 <= t1]
 -- >     where
+-- >     t1                 = head [t | (t, _) <- ee]
 -- >     intervals e        = [(time1, time2, x) | ((time1,x),(time2,_)) <- zip e (tail e)]
 -- >     trim time1 time2 e = [x | (timex,x) <- e, time1 < timex, timex <= time2]
```

Comment:
Neither is correct, I believe, not the least because `t1` is not in scope. 😅  The correct semantics is

```haskell
switchE e0 ee0 time0 =
    concat [ trim t1 t2 e | (t1,t2,e) <- intervals ee ]
  where
    laterThan e time0  = [(timex,x) | (timex,x) <- e, time0 < timex ]
    ee                 = [(time0, e0)] ++ (ee0 `laterThan` time0)
    intervals ee       = [(time1, time2, e) | ((time1,e),(time2,_)) <- zip ee (tail ee)]
    trim time1 time2 e = [x | (timex,x) <- e, time1 < timex, timex <= time2]
```


Formulated in this way, it becomes clearer how `e` aligns with the occurrences in `ee`, and the `trim` function indicates that `<=` instead of `<` is the right choice.

The other main question is: What happens right at `time0`? If the `ee` event has an occurrence at this moment, then this will be ignored — that's why we need the initial event `e` in the first place. Any occurrence of `e` at this moment will also be ignored, however, also by virtue of `trim`.

(That said, the semantics was never quite correct in the case where `ee` is a singleton or the empty list. 🙈 But we can exclude this corner case by proclaiming that the semantics are only supposed to hold in the other cases.)